File: Home/Client/app/components/budget/budget.ts
export interface Budget 
{
    budgetId: number,
    budgetUid: string,
    budgetName: string,
    frequency: TransactionFrequency,
    type: TransactionType,
    createdDate: Date,
    updatedDate: Date,
    amount: number
}

enum TransactionFrequency
{
    weekly = 0,
    biweekly = 1,
    monthly = 2,
    bimonthly = 3,
    semiannually = 4,
    annually = 5,
    none = 6
}

enum TransactionType
{
    credit = 0,
    debit = 1
}

File: Home/Server/Budgets/Budget.cs
using System;
using System.Collections.Generic;

namespace Home.Server.Budgets
{
    public class Budget 
    {
        public int budgetId { get; set; }
        public string budgetName { get; set; }
        public TransactionFrequency frequency { get; set; }
        public TransactionType type { get; set; }
        public DateTime createdDate { get; set; }
        public DateTime updatedDate { get; set; }
        public double amount { get; set; }
        public ICollection<BudgetItem> items { get; set; }
    }
}
